sdk/core/core/src/az_uri.c

```diff
@@ -7,25 +7,22 @@
 
 #include <assert.h>
 #include <stdbool.h>
+#include <ctype.h>
 
 #include <_az_cfg.h>
 
 AZ_NODISCARD AZ_INLINE bool should_encode(uint8_t const c) {
   switch (c) {
-    default:
-      return !(('0' <= c && c <= '9') || ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z'));
     case '-':
     case '_':
     case '.':
     case '~':
       return false;
+    default:
+      return !(('0' <= c && c <= '9') || ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z'));
```

Comment:
you might be able to get in trouble here if the source/execution encoding is something like EBCDIC, and MAYBE ShiftJIS.

Reply:
We put all these assumptions to the https://github.com/Azure/azure-sdk-for-c/blob/master/sdk/core/core/src/az_assumptions.c file.
We also rely on that assumption in az_base64.c.
Not that it can't be done, but should you try to compile on EBCDIC, there will be many other problems: we send HTTP requests, and the other side expects ASCII/UTF8.